service/web/subscriberserver_test.go

```diff
@@ -684,3 +685,85 @@ func TestSubscriberServer_ListProjectSubscriptionsFailure(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Nil(t, res)
 }
+
+func TestSubscriberServer_GetSubscriptionWithExistingName(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockProjectCore := mocks4.NewMockICore(ctrl)
+	brokerStore := mocks.NewMockIBrokerStore(ctrl)
+	subscriptionCore := mocks2.NewMockICore(ctrl)
+	manager := mocks3.NewMockIManager(ctrl)
+	mockCredentialsCore := mocks5.NewMockICore(ctrl)
+	cache := cachemock.NewMockICache(ctrl)
+	server := newSubscriberServer(mockProjectCore, brokerStore, subscriptionCore, mockCredentialsCore, manager, cache)
+	ctx := context.Background()
+	req := &metrov1.GetSubscriptionRequest{
+		Name: "projects/project123/subscriptions/testsub",
+	}
+	sub := &subscription.Model{
+		Name:  "projects/project123/subscriptions/testsub",
+		Topic: "projects/project123/topics/test-topic",
+		PushConfig: &subscription.PushConfig{
+			PushEndpoint: "https://www.razorpay.com/api",
+			Attributes: map[string]string{
+				"test_key": "test_value",
+			},
+			Credentials: &credentials.Model{
+				Username:  "test_user",
+				Password:  "test_pass",
+				ProjectID: "project123",
+			},
+		},
+		AckDeadlineSeconds: 10,
+		DeadLetterPolicy: &subscription.DeadLetterPolicy{
+			DeadLetterTopic:     "test_dlq",
+			MaxDeliveryAttempts: 1,
+		},
+		RetryPolicy: &subscription.RetryPolicy{
+			MinimumBackoff: 20,
+			MaximumBackoff: 300,
+		},
+	}
+	subscriptionCore.EXPECT().Get(gomock.Any(), req.GetName()).Times(1).Return(sub, nil)
+	res, err := server.GetSubscription(ctx, req)
+	assert.NotNil(t, res)
+	assert.Nil(t, err)
+	assert.Equal(t, sub.Name, res.Name)
```

Comment:
can we use reflection for comparison..?

Reply:
@sumitkrzp Keep in mind that while reflections are acceptable in unit tests, we want to avoid using them as part of our actual code. reflections tend to be expensive and should be avoided unless absolutely necessary.

Reply:
@sumitkrzp Cannot see the changes.. Also before committing, please verify it is comparing each and every property defined in the object